sdk/state/state.go

```diff
@@ -37,6 +40,46 @@ type Channel struct {
 	latestUnauthorizedCloseAgreement CloseAgreement
 }
 
+type CloseState int
+
+const (
+	CloseError CloseState = iota - 1
+	CloseNone
+	CloseEarlyClosing // a proposed declTx before fully confirmed is submitted
+	CloseClosing      // latest declTx is submitted
+	CloseNeedsClosing // an earlier declTx is submitted
+	CloseClosed
+)
+
+// CloseState infers the close state from the network sequence number and the expected EI seq num
+func (c *Channel) CloseState() CloseState {
+	latestDeclSequence := txbuild.StartSequenceOfIteration(c.startingSequence, c.latestAuthorizedCloseAgreement.Details.IterationNumber)
+	latestCloseSequence := latestDeclSequence + 1
+
+	latestUnauthorizedDeclSequence := txbuild.StartSequenceOfIteration(c.startingSequence, c.latestUnauthorizedCloseAgreement.Details.IterationNumber)
```

Comment:
This few lines leak some logic (the seq number offsets of the decl and close txs) that otherwise only exists in the txbuild package, so I'm pondering if there's a way we can keep that logic in the txbuild package.

I think one way we can do that is to get the current latest agreement's txs and get the seq number from them. This would be less efficient, but I don't think we need to be worried about efficiency here yet. And efficiency will be less of an issue once we start caching txs, which we'll need to at some point (#199).

Reply:
ya I didn't especially like exporting that either, changed here. A bit less pretty though because it introduces an error possibility in `CloseState()` now
https://github.com/stellar/experimental-payment-channels/pull/232/commits/35568c959e2dc72859dce762fe8c762a0c5fa0f2